src/release/api.d

```diff
@@ -44,26 +44,32 @@ public Configuration getConf ( )
 
     Checks whether the oauth config is setup and starts the setup process if not
 
+    Params:
+        assume_yes = if yes, assume yes to any question
+
 *******************************************************************************/
 
-public void checkOAuthSetup ( )
+public void checkOAuthSetup ( bool assume_yes )
```

Comment:
Hmm I am not sure I like the side-effect in `readYesNoResponse()`.. it could come very unexpected to a new reader/user of the code. Maybe it could be an optional parameter for the function

Reply:
For me it is perfectly acceptable and expected that an interactive function behaves differently given global configuration (because of that, this is what I did in the [git-hub tool](https://github.com/sociomantic-tsunami/git-hub/blob/v1.x.x/git-hub#L127-L171), in case you want to check how confusing, or not, is it).

There are a few differences, looking at it, like there is no *force-yes* but a it will *do nothing* if is not running in a interactive console, but it is similar in spirit.

But my feeling is, every global configuration will always change how functions behave in a way that is not "expected". The whole point of global config is convenience, and I think by using the global config in that `readYesNoResponse()` it makes all much more convenient. Maybe you can rename it as `chose()` or something, maybe the `read` part is what sounds weird if it won't read anything in some circumstances. But if it's just `chose()` or `ask()` for me it makes more sense that that choice can be made by the program itself in some situations.